Fixed release issue #30

alimagdye opened the pull request:
#### Description of Change

<!-- Thank you for your Pull Request. Please describe your change here. -->
Fixed the issue: Release logic misses case where first release was an alpha #30, by doing:

-  **New Function**: Added getReleasesOrUpdateSortedByDate to sort releases by date from oldest to newest.
-  **Usage**: used the new function in src/routes/pr.js to fetch releases, instead of the old one which was sorting alphabetically.
-  **Logic Changes**:  Changed the filtration from choosing nightlies only to focusing on non-nightly prereleases (alpha, beta).
-  **Refactoring**: Updated variable names and comments for clarity and readability.


#### Checklist
<!-- Please confirm the following by changing [ ] to [x]. -->

- [x] This PR was not created with AI. (PRs created mainly with AI will be closed. They waste our team's time. We ban repeat offenders.)
- [x] PR description included and stakeholders cc'd
- [x] relevant API documentation, tutorials, and examples are updated and follow the [documentation style guide](https://github.com/electron/electron/blob/main/docs/development/style-guide.md)
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

dsanders11 commented:
The quality of this PR doesn't meet our standards, and we suspect this PR of being entirely AI-generated. While Electron appreciates and encourages community contributions, its maintainers only have limited bandwidth. AI-generated PRs will not be reviewed and will be closed as invalid.

This PR will now be closed. In the future, raising more PRs with purely AI generated content will result in a organization level ban. Please read [our GSoC 2025 contributor guidance](https://electronhq.notion.site/Electron-Google-Summer-of-Code-2025-Contributor-Guidance-1851459d1bd181ac8004ecb9e031b368), specifically the section on "Project Contributions" and subsection 4.

alimagdye commented:
Hello, @dsanders11 
I wanted to clarify that the code is totally written by me. I provided detailed explanations of my thought process and the logic behind the code in the end of this comment . I work as AI Trainer in Outlier AI, and I improve the AI models in writing a high quality code, so I know what may make you thought it’s AI generated:
-	There are some comments I added when I was documenting the code to understand it, I forgot to remove those comments when I made the pull request. I can remove them back.
-	The writing style in importing the new function I made is different than the old one. This happened because I use prettier extension. I can change it back to the old style.

Also, if there are any specific areas that need improving just tell me!

**My explanation for the code**:
First look at the issue. For PR #40634:
At first it was displaying nightly release.
![image](https://github.com/user-attachments/assets/54b502be-a906-4909-b0a0-680fa6d3b735)

And that because it was based on this method:
![image](https://github.com/user-attachments/assets/0ef7cc62-5d4c-4aee-8e25-d687c0fb8390)
It sorts the releases alphabetically by release tag: first nightly, then beta, then alpha, then stable, which takes O(nlogn).

Then in the `getPRReleaseStatus` function first spread the result in an array which takes O(n) to make a shallow copy then reverse which takes O(n) plus time complexity of the function makes result time complexity O(nlogn + n + n) = O(nlogn) in general.
```
async function getPRReleaseStatus(prNumber) {
  const releases = [...(await getReleasesOrUpdate())].reverse();
  ...
}
```
But, when I tried to access and test the release date I found:  
The releases for PR #40634 sorted in creation date from oldest to newest.
![image](https://github.com/user-attachments/assets/29c54355-297e-42cc-8150-0033e4a91643)
This shows the first prerelease non-nightly (alpha, beta) is: v29.0.0-alpha.1

So I made `getReleasesOrUpdateSortedByDate` function which takes O(nlogn), but it'll take less time in response than the previous one. This function sort from oldest to newest creation date. 
Then I changed the condition of filteration to get only non-nightly prereleases (alpha, beta) as specified in the issue #30.
![image](https://github.com/user-attachments/assets/9c782c50-ce9e-4b8c-9f07-a6beaa9b5340)
Finally I changed the naming of the array and its elements to be readable in the new logic.

Another test on PR #42368 returns v32.0.0-alpha.1 prerelease
![image](https://github.com/user-attachments/assets/3099ba32-1387-4f17-84ce-ea45f7d90b36)
After using my human-made code this release is displayed successfully on the PR lookup page.
